traffic_monitor/manager/opsconfig.go

```diff
@@ -138,12 +139,22 @@ func StartOpsConfigManager(
 
 		// fixed an issue here where traffic_monitor loops forever, doing nothing useful if traffic_ops is down,
 		// and would never logging in again.  since traffic_monitor  is just starting up here, keep retrying until traffic_ops is reachable and a session can be established.
+		backoff, err := util.NewBackoff(cfg.TrafficOpsMinRetryInterval, cfg.TrafficOpsMaxRetryInterval, util.DefaultFactor)
```

Comment:
I think returning an error is probably best for the library to do when given invalid input parameters, but I'm not a huge fan of having to add the extra checks around the retry code. @rob05c what do you think about returning an error but also a usable `Backoff` struct with the sane defaults applied? That way we can still log the error but just state that we're running with defaults applied. Then we can simplify the retry logic by getting rid of the if/else conditionals and the `FixedRetryInterval`.

Reply:
I'm really not a fan of using objects when non-nil error was returned. IMO it's safe practice to always assume all other returned values are undefined, when a non-nil error is returned.

Making a pattern where that's the norm, just feels like really risky behavior, like a really risky habit to develop, that will lead to people writing bugs because the habit leads to accidentally using malformed objects elsewhere after a non-nil return.

In fact, elsewhere in this code, where it assumes the `BackOff` is `nil` when an error is returned, makes me super-nervous. Even if we know that's always the case, someone using any library shouldn't assume that; and even if the library promises it, it's still a dangerous programming habit to develop. IMO it'd be much better practice to manually set `backoff = nil`, even though we know it's nil, just to create safe habits.

It doesn't ever make sense for the config to have min>max. We could avoid the extra conditionals here, by checking for that when the Config is parsed.

Since it's an interface, we could also eliminate the later conditionals, by writing a simple `util.NewConstantBackoff` that can't error. IMO it'd be a useful utility to have anyway. Something like: 

```go
func NewConstantBackoff(d time.Duration) Backoff { return &constantBackOff{d} }
type constantBackOff struct { d time.Duration }
func (b *constantBackOff) BackoffDuration() time.Duration { return b.d }
func (b *constantBackOff) Reset() {}
```

```go
const ConstantBackOffDuration = 30 * time.Second

backoff, err := util.NewBackoff(cfg.TrafficOpsMinRetryInterval, cfg.TrafficOpsMaxRetryInterval, util.DefaultFactor)
if err != nil {
  log.Errorf("possible invalid backoff arguments, will use a fixed sleep interval %+v: %+v", , ConstantBackOffDuration, err)
  backoff = util.NewConstantBackoff(ConstantBackOffDuration)
}
```

How do you feel about that?

Reply:
Good point about the risky habit. I like what you propose better, the `NewConstantBackoff` util as a fallback.

Reply:
@rawlinp and @rob05c made the requested changes and added a NewConstantBackoff()